src/calliope/backend/expression_parser.py

```diff
@@ -953,9 +957,8 @@ def helper_function_parser(
 
 
 def sliced_param_or_var_parser(
-    number: pp.ParserElement,
+    slicer: Iterable[pp.ParserElement],
```

Comment:
Generally trying to move to this more generalised approach of defining iterables that go into `pp.MatchFirst` rather than requiring them all to be defined explicitly and then doing `option_1 | option_2 | ...`.